src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -965,10 +969,12 @@ private StringBuilder getFeedbackSessionResultsForQuestionInCsvFormat(
         }
         
         // add the rows for the possible givers and recipients who have missing responses
-        exportBuilder.append(getRemainingRowsInCsvFormat(fsrBundle, entry, question, questionDetails,
-                                            possibleGiversWithoutResponses, possibleRecipientsForGiver, prevGiver))
-                     .append(Const.EOL + Const.EOL);
+        if (isMissingResponsesShown) {
+            exportBuilder.append(getRemainingRowsInCsvFormat(fsrBundle, entry, question, questionDetails,
+                                                possibleGiversWithoutResponses, possibleRecipientsForGiver, prevGiver));
```

Comment:
Although this isn't a new method call added by you, it would be good to rewrite this as:

```
exportBuilder.append(
        getRemainingRowsInCsvFormat(
                fsrBundle, entry, question, questionDetails,
                possibleGiverWithoutResponses, possibleRecipientsForGiver, prevGiver));
```